src/utils/__tests__/schema-utils.test.ts

```diff
@@ -0,0 +1,1210 @@
+import type { Schema } from 'src/types';
+
+import { setSchemaProperties } from 'src/utils/schema-utils';
+
+describe('setSchemaProperties', () => {
+    let collectionSchema: Schema;
+
+    describe('adds a properties object to the schema', () => {
+        beforeEach(() => {
+            collectionSchema = {
+                $defs: {},
+                $ref: 'flow://connector-schema',
+            };
```

Comment:
In the future I think we could store this off the value as a `const` and then we could use it to spread as we check the output down below. Think it helps highlight what is changing.